rt/frontend/simple/src/main/java/org/apache/cxf/endpoint/dynamic/TypeClassInitializer.java

```diff
@@ -50,6 +50,15 @@
     boolean isFault;
     Bus bus;
 
+    @Deprecated // CXF-8658
+    public TypeClassInitializer(ServiceInfo serviceInfo,
```

Comment:
@garydgregory we cannot do that, it will blow with NPE in the `TypeClassInitializer::createFaultClass`

Reply:
Well, crud :-(

Reply:
There are a few options, the `bus` instance should be taken from somewhere, for example if we bring constructor back, we need to introduce `setBus(Bus b)` method which has to be called, not ideal but a solution nonetheless.

Another option - use the bus from the thread local context using BusFactory::getThreadDefaultBus, the caller context should make sure it is set.

Reply:
or just sthg like (`bus != null? bus.getExtension( ExceptionCreator.class) : new ExceptionCreator()).createExceptionClass(cls);`)

Reply:
@rmannibucau afaik does not exist anymore, `ExceptionCreator` is replaced by `ExceptionClassGenerator` which also needs `bus` ... (because of ASM Helper).

Reply:
Sounds like the bus is passed just for the extensions so replacing it by a `Function<Class<?>, Object>` (real extension "lookup") could make it trivial to replace wen bus is null, no?